### ISSUE TITLE
False positive for 'Type arguments must be declared prior to const arguments'

### ISSUE THREAD

**Issue by zeroeightysix** [intellij-rust/intellij-rust#6454]
Consider the following code:

```rust
#![feature(const_generics)]

fn foo<const C: usize, A>() {}

fn main() {
    foo::<5, bool>();
}
```

The plugin will insist that the order of type and const parameters is wrong (at error level), even though the code compiles correctly:
![image](https://user-images.githubusercontent.com/27009727/100552832-eb07d700-3289-11eb-8da8-38fd34556ec5.png)

Using rustc 1.49.0-nightly (ffa2e7ae8 2020-10-24).
